freud/locality.pyx

```diff
@@ -359,26 +362,26 @@ cdef class NeighborQuery:
             raise RuntimeError("You cannot use the query method unless this "
                                "object was originally constructed with "
                                "reference points")
-        points = freud.common.convert_array(np.atleast_2d(points),
-                                            shape=(None, 3))
+        query_points = freud.common.convert_array(
+            np.atleast_2d(query_points), shape=(None, 3))
 
         return NeighborQueryResult.init(
-            self, points, exclude_ii, r=0, k=k)
+            self, query_points, exclude_ii, r=0, k=num_neighbors)
 
-    def queryBall(self, points, float r, cbool exclude_ii=False):
+    def queryBall(self, query_points, float r_max, cbool exclude_ii=False):
         R"""Query for all points within a distance r of the provided point(s).
 
         Args:
-            points ((:math:`N`, 3) :class:`numpy.ndarray`):
+            query_points ((:math:`N`, 3) :class:`numpy.ndarray`):
                 Points to query for.
             r (float):
                 The distance within which to find neighbors
             exclude_ii (bool, optional):
                 Set this to :code:`True` if pairs of points with identical
                 indices to those in self.points should be excluded. If this is
                 :code:`None`, it will be treated as :code:`True` if
-                :code:`points` is :code:`None` or the same object as
-                :code:`ref_points` (Defaults to :code:`None`).
+                :code:`query_points` is :code:`None` or the same object as
+                :code:`points`. (Default value = :code:`None`).
```

Comment:
It looks like for some reason the exclude_ii parameter defaults to False. Can you change that, update the code to just set it to points == query_points, and then update the docstring to remove the part about being the same object?

Reply:
I think it would be just better to remove the documentation about `points == query_points` check, since probably `convery_array` called on `points` on construction of `NeighborQuery` object would mess up address if we do `points is query_points`.
And `==` check is expensive. 
I think it should just say it defaults to `False`.